src/Libraries/Microsoft.Extensions.AI/ChatCompletion/FunctionInvokingChatClient.cs

```diff
@@ -840,6 +887,19 @@ FunctionResultContent CreateFunctionResultContent(FunctionInvocationResult resul
         return result;
     }
 
+    /// <summary>This method will invoke the function within the try block.</summary>
+    /// <param name="context">The function invocation context.</param>
+    /// <param name="cancellationToken">Cancellation token.</param>
+    /// <returns>The function invocation context and result.</returns>
+    protected virtual async Task<(FunctionInvocationContext context, object? result)> InvokeFunctionCoreAsync(FunctionInvocationContext context, CancellationToken cancellationToken)
+    {
+        _ = Throw.IfNull(context);
+
+        var result = await context.Function.InvokeAsync(context.Arguments, cancellationToken);
+
+        return (context, result);
```

Comment:
Why does this need to return a FunctionInvocationContext?

Reply:
Since context is modified by reference we don't need to send it back.

As the caller `CurrentContext = context;` right before calling this.

Reply:
I don't understand. Why does this method need to return the context at all?

Reply:
we don't.

Reply:
Returning only the `result`, in latest changes.